copy.go

```diff
@@ -54,6 +57,83 @@ type CopyOptions struct {
 	MapRoot func(ctx context.Context, src content.Storage, root ocispec.Descriptor) (ocispec.Descriptor, error)
 }
 
+// getPlatformFromConfig returns a platform object which is made up from the
+// fields in config blob.
+func getPlatformFromConfig(ctx context.Context, src content.Storage, desc ocispec.Descriptor, targetConfigMediaType string) (*ocispec.Platform, error) {
+	if desc.MediaType != targetConfigMediaType {
+		return nil, fmt.Errorf("mismatch MediaType %s: expect %s", desc.MediaType, targetConfigMediaType)
+	}
+
+	rc, err := src.Fetch(ctx, desc)
+	if err != nil {
+		return nil, err
+	}
+	defer rc.Close()
+
+	var platform *ocispec.Platform
+	if err = json.NewDecoder(rc).Decode(&platform); err != nil {
```

Comment:
There is an edge case on golang JSON parsing. If `rc` contains literally `null`, the decode will succeed and `platform` will be set to `nil`.

Reply:
Try this small example:
```go
package main

import (
	"encoding/json"
	"fmt"
	"strings"
)

func main() {
	var a *int

	err := json.NewDecoder(strings.NewReader("null")).Decode(&a)
	if err != nil {
		panic(err)
	}

	fmt.Println(a)
}
```
It outputs
```
<nil>
```

Reply:
ummm, that's interesting. So if rc contains literally null (e.g., image configuration JSON document contains "author": "null"), then the platform will be set to nil no matter the rc has other OS, architecture values or not?

Reply:
Got it. If the platform was set to nil, then the runtime error will be raised when calling the platform.Match func. I could add a check on the returned platform. If the platform is nil, then throw the missing platform error. What do you think?

Reply:
Changed to use platform struct instead of pointer, and also added a check on the err. If the error is io.eof, will return the empty platform then. Added tests for both null config blob and empty config blob cases.
Thanks for pointing this out.